// hawk/package.json.mjs
var name = "hawk";
var description = "HTTP Hawk Authentication Scheme";
var version = "6.0.2";
var author = "Eran Hammer <eran@hammer.io> (http://hueniverse.com)";
var repository = "git://github.com/hueniverse/hawk";
var main = "lib/index.js";
var browser = "dist/browser.js";
var keywords = [
	"http",
	"authentication",
	"scheme",
	"hawk"
];
var engines = {
	node: ">=4.5.0"
};
var dependencies = {
	hoek: "4.x.x",
	boom: "4.x.x",
	cryptiles: "3.x.x",
	sntp: "2.x.x"
};
var devDependencies = {
	"babel-cli": "^6.1.2",
	"babel-preset-es2015": "^6.1.2",
	code: "4.x.x",
	lab: "14.x.x"
};
var babel = {
	presets: [
		"es2015"
	]
};
var scripts = {
	"build-client": "mkdir -p dist; babel lib/browser.js --out-file dist/browser.js",
	prepublish: "npm run-script build-client",
	test: "lab -a code -t 100 -L",
	"test-cov-html": "lab -a code -r html -o coverage.html"
};
var license = "BSD-3-Clause";
var require$$2 = {
	name: name,
	description: description,
	version: version,
	author: author,
	repository: repository,
	main: main,
	browser: browser,
	keywords: keywords,
	engines: engines,
	dependencies: dependencies,
	devDependencies: devDependencies,
	babel: babel,
	scripts: scripts,
	license: license
};

export { author, babel, browser, require$$2 as default, dependencies, description, devDependencies, engines, keywords, license, main, name, repository, scripts, version };
