// http-signature/lib/index.mjs
import { __require as requireParser } from './parser.mjs';
import { __require as requireSigner } from './signer.mjs';
import { __require as requireVerify } from './verify.mjs';
import { __require as requireUtils } from './utils.mjs';

var lib;
var hasRequiredLib;

function requireLib () {
	if (hasRequiredLib) return lib;
	hasRequiredLib = 1;
	// Copyright 2015 Joyent, Inc.

	var parser = requireParser();
	var signer = requireSigner();
	var verify = requireVerify();
	var utils = requireUtils();



	///--- API

	lib = {

	  parse: parser.parseRequest,
	  parseRequest: parser.parseRequest,

	  sign: signer.signRequest,
	  signRequest: signer.signRequest,
	  createSigner: signer.createSigner,
	  isSigner: signer.isSigner,

	  sshKeyToPEM: utils.sshKeyToPEM,
	  sshKeyFingerprint: utils.fingerprint,
	  pemToRsaSSHKey: utils.pemToRsaSSHKey,

	  verify: verify.verifySignature,
	  verifySignature: verify.verifySignature,
	  verifyHMAC: verify.verifyHMAC
	};
	return lib;
}

export { requireLib as __require };
