// fs-extra/lib/move/move-sync.mjs
import { g as gracefulFs } from '../../../graceful-fs/graceful-fs.mjs';
import path$1 from 'path';
import { c as copy } from '../copy/index.mjs';
import { r as remove_1 } from '../remove/index.mjs';
import { m as mkdirs } from '../mkdirs/index.mjs';
import { s as stat$1 } from '../util/stat.mjs';

const fs = gracefulFs;
const path = path$1;
const copySync = copy.copySync;
const removeSync = remove_1.removeSync;
const mkdirpSync = mkdirs.mkdirpSync;
const stat = stat$1;

function moveSync (src, dest, opts) {
  opts = opts || {};
  const overwrite = opts.overwrite || opts.clobber || false;

  const { srcStat, isChangingCase = false } = stat.checkPathsSync(src, dest, 'move', opts);
  stat.checkParentPathsSync(src, srcStat, dest, 'move');
  if (!isParentRoot(dest)) mkdirpSync(path.dirname(dest));
  return doRename(src, dest, overwrite, isChangingCase)
}

function isParentRoot (dest) {
  const parent = path.dirname(dest);
  const parsedPath = path.parse(parent);
  return parsedPath.root === parent
}

function doRename (src, dest, overwrite, isChangingCase) {
  if (isChangingCase) return rename(src, dest, overwrite)
  if (overwrite) {
    removeSync(dest);
    return rename(src, dest, overwrite)
  }
  if (fs.existsSync(dest)) throw new Error('dest already exists.')
  return rename(src, dest, overwrite)
}

function rename (src, dest, overwrite) {
  try {
    fs.renameSync(src, dest);
  } catch (err) {
    if (err.code !== 'EXDEV') throw err
    return moveAcrossDevice(src, dest, overwrite)
  }
}

function moveAcrossDevice (src, dest, overwrite) {
  const opts = {
    overwrite,
    errorOnExist: true,
    preserveTimestamps: true
  };
  copySync(src, dest, opts);
  return removeSync(src)
}

var moveSync_1 = moveSync;

export { moveSync_1 as m };
