// tough-cookie/package.json.mjs
var author = {
	name: "Jeremy Stashewsky",
	email: "jstashewsky@salesforce.com",
	website: "https://github.com/stash"
};
var contributors = [
	{
		name: "Alexander Savin",
		website: "https://github.com/apsavin"
	},
	{
		name: "Ian Livingstone",
		website: "https://github.com/ianlivingstone"
	},
	{
		name: "Ivan Nikulin",
		website: "https://github.com/inikulin"
	},
	{
		name: "Lalit Kapoor",
		website: "https://github.com/lalitkapoor"
	},
	{
		name: "Sam Thompson",
		website: "https://github.com/sambthompson"
	},
	{
		name: "Sebastian Mayr",
		website: "https://github.com/Sebmaster"
	}
];
var license = "BSD-3-Clause";
var name = "tough-cookie";
var description = "RFC6265 Cookies and Cookie Jar for node.js";
var keywords = [
	"HTTP",
	"cookie",
	"cookies",
	"set-cookie",
	"cookiejar",
	"jar",
	"RFC6265",
	"RFC2965"
];
var version = "2.3.4";
var homepage = "https://github.com/salesforce/tough-cookie";
var repository = {
	type: "git",
	url: "git://github.com/salesforce/tough-cookie.git"
};
var bugs = {
	url: "https://github.com/salesforce/tough-cookie/issues"
};
var main = "./lib/cookie";
var files = [
	"lib"
];
var scripts = {
	suffixup: "curl -o public_suffix_list.dat https://publicsuffix.org/list/public_suffix_list.dat && ./generate-pubsuffix.js",
	test: "vows test/*_test.js"
};
var engines = {
	node: ">=0.8"
};
var devDependencies = {
	async: "^1.4.2",
	"string.prototype.repeat": "^0.2.0",
	vows: "^0.8.1"
};
var dependencies = {
	punycode: "^1.4.1"
};
var require$$6 = {
	author: author,
	contributors: contributors,
	license: license,
	name: name,
	description: description,
	keywords: keywords,
	version: version,
	homepage: homepage,
	repository: repository,
	bugs: bugs,
	main: main,
	files: files,
	scripts: scripts,
	engines: engines,
	devDependencies: devDependencies,
	dependencies: dependencies
};

export { author, bugs, contributors, require$$6 as default, dependencies, description, devDependencies, engines, files, homepage, keywords, license, main, name, repository, scripts, version };
