// s-update-manager/lib/src/feature/config/const.mjs
const defaultConfig = {
    snpCatalog: './.snp',
    template: 'node',
    sUpdaterVersion: undefined,
    availableSNPSuffix: ['-default.md', '-instructions.md', '-custom.md', '-extend.md'],
    availableSNPKeySuffix: ['defaultFile', 'instructionsFile', 'customFile', 'extendFile'],
    templateVersion: undefined,
    REPOSITORY_MAP_FILE_NAME: 'repositoryMap.json',
    snpFileMapConfig: './.snp/repositoryMap.json',
    projectCatalog: './',
    temporaryFolder: './.snp/temporary/',
    snpConfigFileName: 'snp.config.json',
    snpConfigFile: './.snp/snp.config.json',
    remoteRepository: 'https://raw.githubusercontent.com/SebastianWesolowski/s-update-manager/dev/template/',
    // remoteRepository: 'https://raw.githubusercontent.com/SebastianWesolowski/s-update-manager/main/template/',
    repositoryUrl: 'https://raw.githubusercontent.com/SebastianWesolowski/s-update-manager/dev/template/node',
    isDebug: false,
    _: [],
};
const defaultTemplateConfig = {
    projectCatalog: './',
    templateCatalogName: 'templateCatalog',
    templateCatalogPath: './templateCatalog',
    repositoryMapFileName: 'repositoryMap.json',
    repositoryMapFilePath: './repositoryMap.json',
    bumpVersion: true,
    isDebug: false,
    _: [],
};

export { defaultConfig, defaultTemplateConfig };
